app/models/observers/notice_observer.rb

```diff
@@ -40,9 +40,19 @@ def plan_year_update(new_model_event)
             end
           end
         end
-      end
 
-      if PlanYear::DATA_CHANGE_EVENTS.include?(new_model_event.event_key)
+        if new_model_event.event_key == :renewal_enrollment_confirmation
+          errors = plan_year.enrollment_errors
+          unless errors.include?(:non_business_owner_enrollment_count) || errors.include?(:eligible_to_enroll_count)
```

Comment:
You don't need this check here.
There is guard on state transition already to check for these errors.